### PR TITLE
SNOW-499495 Add Cursor.query attribute for accessing last query

### LINKED ISSUE #936: SNOW-499495: Add an attribute with the cursor's last executed query
Django's `DatabaseOperations.last_executed_query()` returns the last executed query for a given cursor. While sqlite3 don't provide a way to get the SQL of the last query and thus [parameter escaping has to be done in Django](https://github.com/django/django/blob/25157033e979c134d455d46995a6db0838457d98/django/db/backends/sqlite3/operations.py#L135-L178), MySQL's drivers provides [Cursor._executed](https://github.com/PyMySQL/mysqlclient/blob/8b920774f27d9d17bc85d1593e5d4a36a7ce3f65/MySQLdb/cursors.py#L322), psycopg2 provides [Cursor.query](https://www.psycopg.org/docs/cursor.html#cursor.query), and cx_Oracle provides [Cursor.statement](https://cx-oracle.readthedocs.io/en/latest/api_manual/cursor.html#Cursor.statement). It would be ideal if snowflake-connector-python could offer the same. The following proposal works for the Django tests I've tried so far (the name `_query` could be changed to whatever you like):
```diff
diff --git a/src/snowflake/connector/cursor.py b/src/snowflake/connector/cursor.py
index d238204..5a51137 100644
--- a/src/snowflake/connector/cursor.py
+++ b/src/snowflake/connector/cursor.py
@@ -248,6 +248,7 @@ class SnowflakeCursor:
         self._result_set: Optional["ResultSet"] = None
         self._result_state: ResultState = ResultState.DEFAULT
         self._use_dict_result = use_dict_result
+        self._query = None
         # TODO: self._query_result_format could be defined as an enum
         self._query_result_format: Optional[str] = None
 
@@ -552,6 +553,7 @@ class SnowflakeCursor:
                 {p["name"]: p["value"] for p in parameters}
             )
 
+        self._query = query
         self._sequence_counter = -1
         return ret
 ```

### PR DESCRIPTION
**Pull request by timgraham**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #936

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Should be straightforward.

### CODE REVIEW

File: src/snowflake/connector/cursor.py [hunk 1]
```diff
@@ -274,6 +275,10 @@ def __del__(self) -> None:  # pragma: no cover
     def description(self) -> List[ResultMetadata]:
         return self._description
 
+    @property
+    def query(self) -> Optional[str]:
+        return self._query
```
Comment (sfc-gh-mkeller) [inline src/snowflake/connector/cursor.py:280]: I'm not sure if I agree with making this a property, what do you think about having it as a regular attribute?

Comment (timgraham) [reply src/snowflake/connector/cursor.py:280]: I followed the pattern of other underscore prefixed attribute and property pairs, but as far as I see, this Java-like pattern just adds boilerplate.

Would you also be in favor of removing this pattern elsewhere? It's used quite heavily throughout the project.

Comment (sfc-gh-mkeller) [reply src/snowflake/connector/cursor.py:280]: It makes sense in some cases. I would be for changing existing ones, but let's do that in another PR.

**Review by sfc-gh-jbahk — APPROVED** [review]

**Review by sfc-gh-mkeller — APPROVED** [review]
🚢 🚢

**Pull request merged**